api/path_node.go

```diff
@@ -230,6 +235,17 @@ func CreateHorizonDevice(device *HorizonDevice,
 		return errorhandler(NewSystemError(fmt.Sprintf("error adding architecture for the exchange node. %v", err))), nil, nil
 	}
 
+	// update the node type for the exchange node
+	if *device.NodeType != exchDevice.NodeType {
+		pdr = exchange.PatchDeviceRequest{}
+		pdr.NodeType = *device.NodeType
+		if err := patchDeviceHandler(deviceId, *device.Token, &pdr); err != nil {
```

Comment:
I dont think we should be changing the node type. If there is a node with type device in the exchange, but this API call is for registering for a cluster type, then I think the registration should fail. And we should have a unit test for this case too.